### DISCUSSION TITLE
MSC-Device on a ESP32-S3

### DISCUSSION THREAD

**Discussion by ClaudioFoscan** [adafruit/Adafruit_TinyUSB_Arduino#241] [Q&A]
Hi 

i was wonderingif anyone has a functioning msc-device made with a esp32-s3-wroomer?

I tryed to use the tinyUsb-Example msc-sdfat. so far i got the sd-Card working (Output the directory in the Serialmonitor) but windows device manager wil see a unknown device with errors ...(Error on retriving the device descrition)

what needs to b e done to get it working?

Any special configuration of the Board? or am i missing some configuration stuff in the Source?

im using VS-Code with arduino plugin 0.4.12, the latest board-Libs from Espressif in verison 2.0.6, Adafruit TinyUSB Lib 1.17.1 and sdfat adafruit fork 2.2.1

**Comment by TigoTas** [comment]
Did you manage to get your project going?

This guy, @frankcohen, has an example for a project with ESP32S3 [here](https://github.com/frankcohen/ReflectionsOS)

His HOOBER project was inspired on the Adafruit example.

I wasn't able to get both his code or the one from tinyUSB to work with my board.

I am using a TTGO T3S3.

They compile and upload without a hitch, but the card doesn't mount. I keep getting a "INSERT DISK" message.

My guess is that I have to tinker with sdfat lib, since the board has a non-stardard SPI for the sd card. Still, I have no idea where to start.

![settings](https://user-images.githubusercontent.com/54421252/225121507-67e850f7-13a8-461d-8e9d-8ed092bdab70.jpg)
 
![T3S3](https://user-images.githubusercontent.com/54421252/225121519-c63bd9ab-a6a9-417e-aa21-386adba07836.jpg)

![Result](https://user-images.githubusercontent.com/54421252/225122160-ef8ce243-65c4-461d-915a-da78dd92c649.jpg)

If you, by any chance, got yours working...any help is appreciated.

**Reply by ClaudioFoscan** [reply]
I'm still on it, as soon as i have a solution i'll let you know ...
